### PR TITLE
MAINT: signal: deduplicate *ord functions

### PR DESCRIPTION
**Pull request by ev-br**
Pure maintenance: reduce code duplication in `{butter,cheb1,cheb2,ellip}ord` functions.

### CODE REVIEW

File: scipy/signal/tests/test_filter_design.py
Comment (tylerjereddy) [inline scipy/signal/tests/test_filter_design.py:1796]: minor: should all these added tests have the same name? `cheb2ord` is being compared to `buttord` but `ellip` in still in the test name? Not sure if intentional or copy/paste thing?

Also, was `1.9.1` chosen as a reference just because you happen to have it installed or?

**Review by tylerjereddy — COMMENTED** [review]
Looks like almost 100 lines of code removed with the reduced duplication, which is great. I just added some minor comments after rebasing/fixing some merge conflict locally and running the tests.

File: scipy/signal/tests/test_filter_design.py
Comment (ev-br) [reply scipy/signal/tests/test_filter_design.py]: Copy-paste error, yes. Fixed, thanks!

Re scipy version: absolutely, 1.9.1 is what I had installed at the time, and these things have not been touched for quite a while, so a specific version does not matter all that much I guess.
I've tweaked the comment, too.

File: scipy/signal/tests/test_filter_design.py
Comment (ev-br) [reply scipy/signal/tests/test_filter_design.py]: I don't think we want/need/can/should go after the very last ULP, and these values are not in any sense special, so I've picked this diff into the updated version of this PR.

**Review by larsoner — APPROVED** [review]
Beautiful! At least one current test failure looks related, so once those are fixed +1 for merge from me

**Comment by ev-br** [comment]
Reworked the newly added tests to be less brittle: do not assert specific values of frequencies. These apparently slightly depend on the versions of the dependencies. So only check the filter order to be the same as before the refactor. For cheb{1, 2}ord, the order is the same, so also assert that the frequencies differ.

The remaining CI failures are https://github.com/scipy/scipy/issues/18842.

**Review by ilayn — APPROVED** [review]

**Comment by ilayn** [comment]
Indeed the codebase really breathes now :) Thanks @ev-br

**Comment by tylerjereddy** [comment]
3 core devs happy, and the CI was happy too

**Pull request merged**
